onyo/lib/commands.py

```diff
@@ -866,17 +873,18 @@ def onyo_set(inventory: Inventory,
         If a given path is invalid or changes are made that would result in
         renaming an asset, while `rename` is not true, or if `keys` is empty.
     """
-    paths = paths or []
+    paths = paths or [inventory.root]
     if not keys:
         raise ValueError("At least one key-value pair must be specified.")
-
+    if not all(keys.keys()):
+        raise ValueError("Keys are not allowed to be empty or None-values.")
     if not rename and any(k in inventory.repo.get_asset_name_keys() for k in keys.keys()):
         raise ValueError("Can't change asset name keys without --rename.")
-    if any(k in RESERVED_KEYS for k in keys.keys()):
-        raise ValueError(f"Can't set reserved keys ({', '.join(RESERVED_KEYS)}).")
+    if any(k in RESERVED_KEYS + PSEUDO_KEYS for k in keys.keys()):
+        raise ValueError(f"Can't set reserved keys ({', '.join(RESERVED_KEYS + PSEUDO_KEYS)}).")
```

Comment:
1. See above.
2. Messaging should adjust accordingly. (It's still talking about reserved keys).

Reply:
Aren't pseudo keys reserved keys, too, just as a special case? Or is there another word we have for that?

Reply:
I created the issue #535 to discuss handling of RESERVED_KEYS and PSEUDO_KEYS further, thanks for your feedback!

Reply:
> Aren't pseudo keys reserved keys, too, just as a special case? Or is there another word we have for that?

Not sure, I parse that. We have a word for that: "pseudo keys" ;-)

`RESERVED_KEYS` have different definition than `PSEUDO_KEYS`. And it's kinda different concepts. Reserved keys are for specifying properties that are not recorded as key-value. That's `template` for example.
Whereas pseudo keys are key-value pairs of an asset (and can be queried as such), but the values aren't defined arbitrarily in the YAML but reflect information retrieved from file system or git. That's `path` currently.